Standards/scs-0212-v1-requirements-for-container-registries.md

```diff
@@ -0,0 +1,124 @@
+- Authentication of users
+  - support for multiple authentication systems (IdM integration)
+  - user and user account management
+- Authorization
+  - role-based access control to ensure strict access controls
+- Automation
+  - integration with CI/CD tools e.g. via webhooks
+- Vulnerability scanning
+  - reveal security vulnerabilities in container images
+- Content Trust and Validation
+  - verify image authenticity before running
+  - image signing
+- Multi-tenancy
+  - container registry is able to serve multiple tenants (projects, teams, namespaces)
+  - implementation on the storage level, see e.g. [Keppel](https://github.com/sapcc/keppel#overview)
+- Backup and restore
+  - possible strategies for disaster recovery and data migration scenarios
+- Monitoring
+  - observability is a key feature for operating a service in production so the container registry should expose key metrics
+- HA mode
+  - enable the possibility of system uptime, even if a failure of some sort could occur
+- Registry replication
+  - replication allows users to replicate container images between registries of the same instances and between registries of different instances as well
+- Proxy cache (pull-through cache)
+  - Proxy cache enables the use of a container registry to proxy and cache images from a target public or private registry
+- Quota management
+  - ability to control resource use of components or the whole registry
+- Garbage collection
+  - removing blobs from the filesystem when they are no longer referenced by a manifest
+- Retention policy
+  - reduce the number of image tags, many of which might not be required after a given time or once a subsequent image tag has superseded them
+
+#### Desirable features
+
+- Additionally supported artifacts
+  - Additional artifacts that the registry is able to store in addition to OCI artifacts, e.g. Java, Node.js, or Python packages
+- Integration possibilities
+  - Ability to cooperate with another software solution in order to improve own feature set (e.g. integration of P2P solution for improving container image distribution (download speed and stability, high scalability, ...))
+- Deployment capabilities
+  - ways to deploy a container registry (only "official" ways are mentioned)
+- Administration capabilities "as a code"
+  - ability to manage container registry via "as a code" solutions, e.g. Ansible role
+
+## Standard
+
+It is very unlikely, that every Container registry can fulfill every requirement defined by this document, and probably
+not all deployments require every feature listed here. The required feature set should therefore be carefully considered
+by the provider of the registry. We nevertheless RECOMMEND using most of the feature set provided in this document.
+
+If the features mentioned above are all considered, a possible registry solution SHOULD fulfill the required
+features to 90% or more. But this is again dependent on the specific Use case and the surrounding architecture.
```

Comment:
```suggestion
features to 90% or more. But this is again dependent on the specific use case and the surrounding architecture.
```

Reply:
Changed in https://github.com/SovereignCloudStack/standards/pull/458/commits/e73ed463c4d03e128dcfa7f1437bfa7131de3f93